### PR TITLE
Replace Range<T> with StateTransition<T>

### LINKED ISSUE #6645: Why `Range<BufferUses>`
What is going on with `BufferBarrier`/`TextureBarrier` `usage: Range<BufferUses>` and `usage: Range<TextureUses>`. This smells real fishy, ranges of bitfields dont sound like they should work. They're not ordered, whats this all about?

### PR DESCRIPTION
**Pull request by atlv24**
**Connections**
Fixes #6645

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Review by teoxoy — APPROVED** [review]
Much clearer, thanks!

**Comment by teoxoy** [comment]
Not sure why CI is failing, I went through each change and everything seemed in order (no change in behavior).

**Review by ErichDonGubler — APPROVED** [review]
yesplz

File: wgpu-core/src/resource.rs [hunk 1]
```diff
@@ -636,11 +636,17 @@ impl Buffer {
...
+                            to: hal::BufferUses::COPY_SRC,
+                        },
                     };
                     let transition_dst = hal::BufferBarrier::<dyn hal::DynBuffer> {
                         buffer: raw_buf,
-                        usage: hal::BufferUses::empty()..hal::BufferUses::COPY_DST,
+                        usage: hal::StateTransition {
+                            from: hal::BufferUses::empty(),
+                            to: hal::BufferUses::COPY_SRC,
+                        },
```
Comment (ErichDonGubler) [inline wgpu-core/src/resource.rs]: issue: Found the bug! 😀

Comment (ErichDonGubler) [reply wgpu-core/src/resource.rs]: Solution: s/SRC/DST in the "new" portion of the diff.

Comment (ErichDonGubler) [reply wgpu-core/src/resource.rs]: Pushed a fix with 8e8d04f74c791f9f39deb29f93385c05dfc25db4.

Comment (teoxoy) [reply wgpu-core/src/resource.rs]: Ah! Good catch!

**Pull request merged**